src/main/java/com/google/sps/servlets/SavedRecipeServlet.java

```diff
@@ -64,9 +64,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Long servings = (Long) entity.getProperty("servings");
       Long readyInMinutes = (Long) entity.getProperty("readyInMinutes");
       ArrayList<String> dietaryNeedsStrings = (ArrayList<String>) entity.getProperty("dietaryNeeds");
+      ArrayList<String> ingredientNamesStrings = (ArrayList<String>) entity.getProperty("ingredientNames");
       
       // convert string to Diet enum because datastore stores dietaryNeeds as a list of strings
       ArrayList<SavedRecipe.Diet> dietaryNeeds = new ArrayList<>();
+      if(dietaryNeedsStrings == null) {
+        dietaryNeedsStrings = new ArrayList<String>();
+      }
```

Comment:
I think it may make sense to define a helper function to extract property and return an empty list if the property value is null. 
That may also be useful for ingredientNamesStrings, right?